includes/Modules/Analytics_4/Conversion_Reporting/Conversion_Reporting_Events_Sync.php

```diff
@@ -118,11 +128,26 @@ public function sync_detected_events() {
 			$detected_events[] = $row['dimensionValues'][0]['value'];
 		}
 
+		$this->maybe_update_new_and_lost_events( $detected_events, $saved_detected_events );
+
+		$this->settings->merge( array( 'detectedEvents' => $detected_events ) );
+	}
+
+	/**
+	 * Saves new and lost events transients.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $detected_events       Currently detected events array.
+	 * @param array $saved_detected_events Previously saved detected events array.
+	 */
+	protected function maybe_update_new_and_lost_events( $detected_events, $saved_detected_events ) {
 		$new_events  = array_diff( $detected_events, $saved_detected_events );
 		$lost_events = array_diff( $saved_detected_events, $detected_events );
 
 		if ( ! empty( $new_events ) ) {
 			$this->transients->set( self::DETECTED_EVENTS_TRANSIENT, array_values( $new_events ) );
+			$this->new_badge_events_sync->sync_new_events_badge( $new_events );
```

Comment:
@zutigrm this is not a deal breaker for me, but can be convoluted to read when we swap `event` and `badge` interchangeably. Should we perhaps change `sync_new_events_badge` to rather be `sync_new_badge_events`? And of course change the method name ;)

WDYT?

Reply:
Thanks, that makes sense